### PR TITLE
Add from scratch

### PR DESCRIPTION
**Pull request by Jkwok2018**
- Engagement, Test Plan, Test Case, and Test can all add from scratch and be edited in their corresponding details page
- Note that the add endpoints only support **adding from scratch**
- separate cloning endpoints should be implemented but would most likely call these add endpoints as helper 

**Important design note** for the **edit** Engagement/TestPlan/TestCase/Test endpoints:
- they currently only edit everything that are textfields or dropdown (e.g. name, description, etc.)
- it does NOT (and should not imo) handle nested things (e.g. the BOM of a testCase)
- If user wants to edit the nested level, they can do so directly using the corresponding endpoint
                - e.g. if they want to edit BOM of a test case, there should be a separate endpoint _editBOM_ (not implemented yet), not via _editTestCase_

### CODE REVIEW

**Comment by Jkwok2018** [comment]
Cloning:
- `cloneTestCase` and `cloneTestPlan` are suppose to work (endpoint can be found in util/cloneEntry)
- **Note:** made the assumption that you can ONLY clone from test cases or test plans that are in the **libraries**. In addition, if you clone a test plan from the test plan library, we assume that the its children test cases should all be in the test case library

setTestPlanActive:
- the _Set Active_ button in engagement details page should work. i.e. you can set an existing archived test plan to active.
- it calls the new api endpoint `setTestPlanActive`

**Comment by McGarvs** [comment]
The Test Case Details Edit Button is broken / does not populate fields. Not going to test what it does.

**Comment by McGarvs** [comment]
Editing for TestCases and TestPlans is broken, likely because of the changes for cloning in TestCaseModalForm / TestPlanModalForm.

**Comment by Jkwok2018** [comment]
> Editing for TestCases and TestPlans is broken, likely because of the changes for cloning in TestCaseModalForm / TestPlanModalForm.

Should be fixed!

**Comment by Jkwok2018** [comment]
> From what I can tell, isActive is also pretty broken. It shouldn't appear in the modals imo because of the bugs it causes. I also think adding a new TestPlan should default to notActive because of bugs...

- removed isActive field in test plan modal; default to false when creating from scratch or cloning

File: cradlepoint/pages/api/cloneTestCase.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {cloneTestCase} from "../../util/cloneEntry"
+
+export default async function handler(req, res) {
+    if (req.method !== 'POST') {
+      res.status(405).send({ message: 'Only POST requests allowed' })
+    }
+    else {
+      // try {
```
[the remark this answers is not in the document]
Comment (Jkwok2018) [reply cradlepoint/pages/api/cloneTestCase.js]: just uncommented!

**Pull request merged**